servicetests/tests/ShadowUpdate/main.cpp

```diff
@@ -0,0 +1,331 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/crt/Api.h>
+#include <aws/crt/JsonObject.h>
+#include <aws/crt/UUID.h>
+#include <aws/crt/io/HostResolver.h>
+
+#include <aws/crt/mqtt/Mqtt5Packets.h>
+#include <aws/iot/Mqtt5Client.h>
+#include <aws/iot/MqttClient.h>
+
+#include <aws/iotshadow/ErrorResponse.h>
+#include <aws/iotshadow/IotShadowClient.h>
+#include <aws/iotshadow/ShadowDeltaUpdatedEvent.h>
+#include <aws/iotshadow/ShadowDeltaUpdatedSubscriptionRequest.h>
+#include <aws/iotshadow/UpdateNamedShadowRequest.h>
+#include <aws/iotshadow/UpdateShadowRequest.h>
+#include <aws/iotshadow/UpdateShadowResponse.h>
+#include <aws/iotshadow/UpdateShadowSubscriptionRequest.h>
+
+#include <aws/iotshadow/GetShadowRequest.h>
+#include <aws/iotshadow/GetShadowResponse.h>
+#include <aws/iotshadow/GetShadowSubscriptionRequest.h>
+#include <aws/iotshadow/UpdateShadowSubscriptionRequest.h>
+
+#include <algorithm>
+#include <condition_variable>
+#include <iostream>
+#include <mutex>
+#include <thread>
+
+#include "../../../samples/utils/CommandLineUtils.h"
+
+using namespace Aws::Crt;
+using namespace Aws::Iotshadow;
+
+void changeShadowValue(Aws::Crt::String thingName, String property, String value);
+void changeNamedShadowValue(String thingName, String property, String value, String shadowName);
+
+IotShadowClient *shadowClient = nullptr;
+int main(int argc, char *argv[])
+{
+    printf("Starting shadow update test\n");
+    /************************ Setup ****************************/
+    ApiHandle apiHandle;
+
+    /**
+     * cmdData is the arguments/input from the command line placed into a single struct for
+     * use in this sample. This handles all of the command line parsing, validating, etc.
+     * See the Utils/CommandLineUtils for more information.
+     */
+    Utils::cmdData cmdData = Utils::parseSampleInputShadow(argc, argv, &apiHandle);
+
+    /**
+     * In a real world application you probably don't want to enforce synchronous behavior
+     * but this is a sample console application, so we'll just do that with a condition variable.
+     */
+    std::promise<bool> connectionCompletedPromise;
+    std::promise<void> connectionClosedPromise;
+    std::shared_ptr<Aws::Crt::Mqtt::MqttConnection> connection;
+
+    Aws::Iot::Mqtt5ClientBuilder *builder = Aws::Iot::Mqtt5ClientBuilder::NewMqtt5ClientBuilderWithMtlsFromPath(
+        cmdData.input_endpoint, cmdData.input_cert.c_str(), cmdData.input_key.c_str());
+
+    // Check if the builder setup correctly.
+    if (builder == nullptr)
+    {
+        printf(
+            "Failed to setup mqtt5 client builder with error code %d: %s", LastError(), ErrorDebugString(LastError()));
+        return -1;
+    }
+    // Create Mqtt5Client
+    std::shared_ptr<Aws::Crt::Mqtt5::Mqtt5Client> client5;
```

Comment:
Same here, if 3 and 5 are both supported then why put a 5 client out here?

Reply:
I need that client to call stop connection for mqtt5
and need the connection pointer to call close connection for mqtt3